pulsar/resource_pulsar_topic.go

```diff
@@ -165,6 +195,23 @@ func resourcePulsarTopicRead(d *schema.ResourceData, meta interface{}) error {
 		setPermissionGrant(d, grants)
 	}
 
+	if retPoliciesCfg, ok := d.GetOk("retention_policies"); ok && retPoliciesCfg.(*schema.Set).Len() > 0 {
+		if topicName.IsPersistent() {
+
+			ret, err := client.GetRetention(*topicName, true)
+			if err != nil {
+				return fmt.Errorf("ERROR_READ_TOPIC: GetRetention: %w", err)
+			}
+
+			_ = d.Set("retention_policies", schema.NewSet(retentionPoliciesToHash, []interface{}{
+				map[string]interface{}{
+					"retention_minutes":    fmt.Sprint(ret.RetentionTimeInMinutes),
+					"retention_size_in_mb": fmt.Sprint(ret.RetentionSizeInMB),
```

Comment:
Looks `retention_size` and `retention_time` is more reasonable here, looks like 

```
retention_size = "1G"
retention_time = "7d"
```

Reply:
It same with `pulsar_namespace`, should we update the `pulsar_namespace` config?

Reply:
I tried this way, this data conversion is difficult. We can set `1G` in Terraform configuration file and parse this value into Pulsar, when we get this policy from the pulsar, `1024` is returned, which is used to apply to Terraform configuration file, but we cannot determine the suffix, so it is difficult.

Reply:
@nodece I see. It's not a big problem.